### PR TITLE
Patch for liquibase-core && liquibase-cli

### PR DESCRIPTION
**Pull request by EggOxygen**

### CODE REVIEW

**Comment by EggOxygen** [comment]
Ignore these 'Build and Test', XD.

File: liquibase-core/pom.xml [hunk 1]
```diff
@@ -270,19 +270,19 @@
...
+<!--                <configuration>-->
+<!--                    <skip>true</skip>-->
+<!--                </configuration>-->
+<!--                <executions>-->
+<!--                    <execution>-->
+<!--                        <id>default-install</id>-->
+<!--                        <phase>none</phase>-->
+<!--                    </execution>-->
+<!--                </executions>-->
+<!--            </plugin>-->
```
Comment (unlimitedsola) [inline liquibase-core/pom.xml:285]: We can use block comments to save a few diff lines.

File: liquibase-core/src/main/java/liquibase/command/core/DiffChangelogCommandStep.java [hunk 2]
```diff
@@ -4,6 +4,7 @@
 import liquibase.configuration.ConfigurationValueObfuscator;
 import liquibase.exception.CommandExecutionException;
 
+@SuppressWarnings("all") // third-party code
```
Comment (unlimitedsola) [inline liquibase-core/src/main/java/liquibase/command/core/DiffChangelogCommandStep.java:7]: Don't need this anymore as we are maintaining in a separated repository now

File: liquibase-core/src/main/java/liquibase/command/core/InternalDiffChangelogCommandStep.java [hunk 3]
```diff
@@ -11,18 +11,26 @@
 
 import java.io.PrintStream;
 
+@SuppressWarnings("all") // third-party code
```
Comment (unlimitedsola) [inline liquibase-core/src/main/java/liquibase/command/core/InternalDiffChangelogCommandStep.java:14]: Don't need this anymore as we are maintaining in a separated repository now

File: liquibase-core/src/main/java/liquibase/command/core/InternalGenerateChangelogCommandStep.java [hunk 4]
```diff
@@ -17,6 +17,7 @@
 import java.io.File;
 import java.io.PrintStream;
 
+@SuppressWarnings("all") // third-party code
```
Comment (unlimitedsola) [inline liquibase-core/src/main/java/liquibase/command/core/InternalGenerateChangelogCommandStep.java:20]: Don't need this anymore as we are maintaining in a separated repository now

File: liquibase-core/src/main/java/liquibase/integration/commandline/CommandLineUtils.java [hunk 5]
```diff
@@ -36,6 +35,7 @@
  *
  * @author Peter Murray
  */
+@SuppressWarnings("all") // third-party code
```
Comment (unlimitedsola) [inline liquibase-core/src/main/java/liquibase/integration/commandline/CommandLineUtils.java:38]: Don't need this anymore as we are maintaining in a separated repository now

File: liquibase-core/src/main/java/liquibase/integration/commandline/Main.java [hunk 6]
```diff
@@ -52,6 +52,7 @@
  *
  * @deprecated use liquibase.integration.commandline.LiquibaseCommandLine.
  */
+@SuppressWarnings("all") // third-party code
```
Comment (unlimitedsola) [inline liquibase-core/src/main/java/liquibase/integration/commandline/Main.java:55]: Don't need this anymore as we are maintaining in a separated repository now

File: liquibase-core/src/main/java/liquibase/command/core/InternalGenerateChangelogCommandStep.java [hunk 7]
```diff
@@ -25,14 +26,14 @@ public class InternalGenerateChangelogCommandStep extends InternalDiffChangelogC
             "BEST PRACTICE: When generating formatted SQL changelogs, always check if the 'splitStatements' attribute" + System.lineSeparator() +
             "works for your environment. See https://docs.liquibase.com/commands/generatechangelog.html for more information. ";
 
-    public static final CommandArgumentDefinition<String> AUTHOR_ARG;
-    public static final CommandArgumentDefinition<String> CONTEXT_ARG;
+    public static final CommandArgumentDefinition<Boolean> OVERWRITE_OUTPUT_FILE_ARG;
```
Comment (unlimitedsola) [inline liquibase-core/src/main/java/liquibase/command/core/InternalGenerateChangelogCommandStep.java:29]: Don't remember we needed this, the old lines indeed need to be deleted, but I don't think we needed this `OVERWRITE_OUTPUT_FILE_ARG` property

File: liquibase-core/src/main/java/liquibase/integration/commandline/CommandLineUtils.java [hunk 8]
```diff
@@ -97,17 +97,6 @@ public static Database createDatabaseObject(ResourceAccessor resourceAccessor,
...
-                // Get values from the configuration object if they aren't already set
-                //
-                if (liquibaseCatalogName == null) {
-                    ConfiguredValue<String> configuredValue  = GlobalConfiguration.LIQUIBASE_CATALOG_NAME.getCurrentConfiguredValue();
-                    liquibaseCatalogName = configuredValue.getValue();
-                }
-                if (liquibaseSchemaName == null) {
-                    ConfiguredValue<String> configuredValue  = GlobalConfiguration.LIQUIBASE_SCHEMA_NAME.getCurrentConfiguredValue();
-                    liquibaseSchemaName = configuredValue.getValue();
-                }
```
Comment (unlimitedsola) [inline liquibase-core/src/main/java/liquibase/integration/commandline/CommandLineUtils.java:110]: is this intentional?

Comment (EggOxygen) [reply liquibase-core/src/main/java/liquibase/integration/commandline/CommandLineUtils.java:110]: It seems that our patch removes this part, I found that this part was added at https://github.com/ocpddev/liquibase/commit/b86e910635e656ac872d8a933953e93ff3c2187e. Now i add it back.

File: liquibase-core/src/main/java/liquibase/command/core/DiffChangelogCommandStep.java [hunk 9]
```diff
@@ -68,14 +68,12 @@ public class DiffChangelogCommandStep extends AbstractCliWrapperCommandStep {
             .description("Include the tablespace attribute in the changelog").build();
         SCHEMAS_ARG = builder.argument("schemas", String.class)
                 .description("Schemas to include in diff").build();
-        INCLUDE_SCHEMA_ARG = builder.argument("includeSchema", Boolean.class)
-                .defaultValue(false)
-                .description("If true, the schema will be included in generated changeSets").build();
-        INCLUDE_CATALOG_ARG = builder.argument("includeCatalog", Boolean.class)
-                .defaultValue(false)
-                .description("If true, the catalog will be included in generated changeSets").build();
```
Comment (unlimitedsola) [inline liquibase-core/src/main/java/liquibase/command/core/DiffChangelogCommandStep.java:76]: Probably not needed

File: liquibase-cli/pom.xml [hunk 10]
```diff
@@ -3,20 +3,25 @@
...
     <artifactId>liquibase-cli</artifactId>
+    <packaging>jar</packaging>
 
     <dependencies>
         <dependency>
-            <groupId>org.liquibase</groupId>
+            <groupId>dev.ocpd.liquibase</groupId>
             <artifactId>liquibase-core</artifactId>
-            <version>${project.version}</version>
+            <version>4.18.0-SNAPSHOT</version>
```
Comment (unlimitedsola) [inline liquibase-cli/pom.xml:24]: revert this

**Pull request merged**